libraries/plugins/elasticsearch/elasticsearch_plugin.cpp

```diff
@@ -291,15 +364,22 @@ void elasticsearch_plugin::plugin_set_program_options(
          ("elasticsearch-node-url", boost::program_options::value<std::string>(), "Elastic Search database node url")
          ("elasticsearch-bulk-replay", boost::program_options::value<uint32_t>(), "Number of bulk documents to index on replay(5000)")
          ("elasticsearch-bulk-sync", boost::program_options::value<uint32_t>(), "Number of bulk documents to index on a syncronied chain(10)")
-         ("elasticsearch-logs", boost::program_options::value<bool>(), "Log bulk events to database")
          ("elasticsearch-visitor", boost::program_options::value<bool>(), "Use visitor to index additional data(slows down the replay)")
+         ("elasticsearch-basic-auth", boost::program_options::value<std::string>(), "Pass basic auth to elasticsearch database ")
+         ("elasticsearch-index-prefix", boost::program_options::value<std::string>(), "Add a prefix to the index(bitshares-)")
          ;
    cfg.add(cli);
 }
 
 void elasticsearch_plugin::plugin_initialize(const boost::program_options::variables_map& options)
 {
-   database().applied_block.connect( [&]( const signed_block& b){ my->update_account_histories(b); } );
+   database().applied_block.connect( [&]( const signed_block& b) {
+      if(!my->update_account_histories(b))
+      {
+         edump(("Error populating ES database, exiting to avoid history gaps."));
+         throw;
```

Comment:
Need to throw graphene::chain::plugin_exception if you want to re-apply the block.